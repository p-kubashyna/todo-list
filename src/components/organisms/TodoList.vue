<script setup>
import TodoElement from '@/components/molecules/TodoElement.vue'
import { useStore } from 'vuex'
import draggable from 'vuedraggable'
import { computed } from 'vue'

const store = useStore()
const todos = computed(() => store.getters.getTodoList)
</script>

<template>
  <div>
    <draggable :list="todos" itemKey="id">
      <template #item="{element}">
        <TodoElement v-bind:todo-item="element" v-bind:key="element.id"/>
      </template>
    </draggable>
  </div>
</template>
