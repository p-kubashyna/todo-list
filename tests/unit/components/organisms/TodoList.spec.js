import { describe, expect, it } from "@jest/globals";
import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import TodoList from "@/components/organisms/TodoList.vue";

describe("Given the TodoList component", () => {
  const todos = [
    { name: "todo-1", isCompleted: false, id: "id-1" },
    { name: "todo-2", isCompleted: true, id: "id-2" },
  ];

  it("should render", () => {
    const store = createStore({
      getters: {
        getTodoList: () => todos,
      },
    });
    const wrapper = mount(TodoList, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
