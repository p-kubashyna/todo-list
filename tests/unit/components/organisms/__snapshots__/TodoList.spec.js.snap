// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the TodoList component should render 1`] = `
<div>
  <div>
    <div
      class="item"
      data-draggable="true"
    >
      <input
        class="checkbox-round"
        type="checkbox"
      />
      <span
        class="task"
      >
        todo-1
      </span>
      <button
        class="delete-button"
      >
        <img
          alt="delete icon"
          class="delete-button-image"
          src=""
        />
      </button>
    </div>
    <div
      class="item"
      data-draggable="true"
    >
      <input
        checked=""
        class="checkbox-round"
        type="checkbox"
      />
      <span
        class="task completed-task"
      >
        todo-2
      </span>
      <button
        class="delete-button"
      >
        <img
          alt="delete icon"
          class="delete-button-image"
          src=""
        />
      </button>
    </div>
  </div>
</div>
`;
